src/Main.lhs

```diff
@@ -202,6 +206,84 @@ Add any special options or imports required by the parsing machinery.
 >                    )
 >       in
 
+-------------------------------------
+Branch off to continuation-based LALR parser production:
+
+>       getForallMatch cli >>= \forallMatch ->
+>       let showTypes = OptCB_ShowTypes `elem` cli || rank2Types
+>           showComments = OptCB_ShowComments `elem` cli
+>           rank2Types = maybe False (return True) forallMatch
+>           match = fromMaybe "" forallMatch
+>           rulesTupleBased = OptCB_RAD_TupleBased `elem` cli
+>       in
+
+>       if OptCB_LALR `elem` cli
+>       then
+>       let (isMonad, _, parserType, _, _) = monad g
+>       
+>           ptype = case (Grammar.lexer g, isMonad) of
+>             (Nothing, False) -> LALR.Normal
+>             (Nothing, True) -> LALR.Monad
+>             (Just _, False) -> error "%lexer without %monad not supported in RAD"
+>             (Just _, True) -> LALR.MonadLexer
+>         
+>           options = LALR.GenOptions {
+>             LALR.ptype = ptype,
+>             LALR.wrapperType = if parserType == "Parser" then "HappyP" else "Parser",
```

Comment:
Well, this can't be right, can it? The info you seek is encoded in the `Grammar`'s `monad` field.

Reply:
No, this wrapperType is a wrapper _around_ the user-supplied monad type (see 4.2.1 in [here](https://pp.ipd.kit.edu/uploads/publikationen/knothe20bachelorarbeit.pdf): the user-defined monad is called `P`, and we define a wrapper and call it `Parser`). This wrapper is always called "Parser", except when the `Grammar`'s `monad` field is also called "Parser"; then we call the wrapper type "HappyP".